python/res/enkf/ert_workflow_list.py

```diff
@@ -2,34 +2,85 @@
     _has_workflow   = ResPrototype("bool ert_workflow_list_has_workflow(ert_workflow_list, char*)")
     _get_workflow   = ResPrototype("workflow_ref ert_workflow_list_get_workflow(ert_workflow_list, char*)")
+    _add_workflow   = ResPrototype("workflow_ref ert_workflow_list_add_workflow(ert_workflow_list, char*, char*)")
     _get_context    = ResPrototype("subst_list_ref ert_workflow_list_get_context(ert_workflow_list)")
     _add_job        = ResPrototype("void ert_workflow_list_add_job(ert_workflow_list, char*, char*)")
     _has_job        = ResPrototype("bool ert_workflow_list_has_job(ert_workflow_list, char*)")
     _get_job        = ResPrototype("workflow_job_ref ert_workflow_list_get_job(ert_workflow_list, char*)")
     _get_job_names  = ResPrototype("stringlist_obj ert_workflow_list_get_job_names(ert_workflow_list)")
-    _free           = ResPrototype("void ert_workflow_list_free(ert_workflow_list)")
 
-    def __init__(self,  ert_workflow_list, config_content):
-        c_ptr = self._alloc(ert_workflow_list, config_content)
+    def __init__(self, subst_list=None, config_content=None, config_dict=None):
+        if subst_list is None:
+            raise ValueError('Failed to construct ErtWorkflowList with no substitution list')
+
+        if config_content is None and config_dict is None:
+            raise ValueError('Failed to construct ErtWorkflowList instance with no config object')
+
+        if config_content is not None and config_dict is not None:
+            raise ValueError('Failed to construct ErtWorkflowList instance with multiple config object')
+
+        c_ptr = None
+
+        if config_content is not None:
+            c_ptr = self._alloc(subst_list, config_content)
+
+        if config_dict is not None:
+            workflow_joblist = WorkflowJoblist()
+            parser = WorkflowJob.configParser( )
+            for job in config_dict.get(ConfigKeys.LOAD_WORKFLOW_JOB, []):
+                try:
+                    new_job = WorkflowJob.fromFile(config_file=job['PATH'], name=job['NAME'], parser=parser)
+                except:
+                    print("WARNING: Unable to create job from {}".format(job['PATH']))
+                    continue
+                new_job.convertToCReference(None)
+                workflow_joblist.addJob(new_job)
+
+            for job_path in config_dict.get(ConfigKeys.WORKFLOW_JOB_DIRECTORY, []) :
+                if not os.path.isdir(job_path):
+                    print("WARNING: Unable to open job directory {}".format(job_path))
+                    continue
+
+                files = os.listdir(job_path)
+                for file_name in files:
+                    full_path = os.path.join(job_path, file_name)
+                    try:
+                        new_job = WorkflowJob.fromFile(config_file=full_path, parser=parser)
+                        new_job.convertToCReference(None)
+                        workflow_joblist.addJob(new_job)
```

Comment:
It's just a minor thing, but it would be "safer" to first add the job, then convert it to a C reference. (same is true for lines 48-49 and 67-69)